src/extractor/guidance/sliproad_handler.cpp

```diff
@@ -29,273 +32,720 @@ SliproadHandler::SliproadHandler(const IntersectionGenerator &intersection_gener
 
-    if (next_road_next_intersection.empty())
+    if (!next)
         return intersection;
 
     // If we are at a traffic loop at the end of a road, don't consider it a sliproad
-    if (intersection_node_id == next_intersection_node)
+    if (intersection_node_id == next->node)
         return intersection;
 
-    std::unordered_set<NameID> target_road_names;
+    std::vector<NameID> target_road_name_ids;
+    target_road_name_ids.reserve(next->intersection.size());
 
-    for (const auto &road : next_road_next_intersection)
+    for (const auto &road : next->intersection)
     {
         const auto &target_data = node_based_graph.GetEdgeData(road.eid);
-        target_road_names.insert(target_data.name_id);
+        target_road_name_ids.push_back(target_data.name_id);
     }
 
-    for (auto &road : intersection)
+    // TODO: remove geojson debugging
+    NodeID b, c, d;
+    b = intersection_node_id;
+    c = next->node;
+
+    auto sliproad_found = false;
+
+    // Check all roads for Sliproads and assign appropriate TurnType
+    for (std::size_t road_index = 0, last = intersection.size(); road_index < last; ++road_index)
     {
-        if (linkTest(road))
+        auto &sliproad = intersection[road_index]; // this is what we're checking and assigning to
+
+        const auto &sliproad_data = node_based_graph.GetEdgeData(sliproad.eid);
+
+        // Discard service and other low priority roads - never Sliproad candidate
+        if (sliproad_data.road_classification.IsLowPriorityRoadClass())
+            continue;
+
+        // This is what we know so far:
+        //
+        //       .
+        //       e
+        //       .
+        //       .
+        // a ... b .... c .   < `next` is intersection at `c`
+        //       `      .
+        //         `    .
+        //           `  .
+        //              d     < `target_intersection` is intersection at `d`
+        //              .       `sliproad_edge_target` is node `d`
+        //              e
+        //
+        //
+        //          ^ `sliproad` is `bd`
+        //       ^ `intersection` is intersection at `b`
+
+        if (!is_potential_link(sliproad))
+            continue;
+
+        // b -> d edge id
+        EdgeID sliproad_edge = sliproad.eid;
+
+        // TODO: remove debugging
+        d = node_based_graph.GetTarget(sliproad_edge);
+
+        const auto target_intersection = [&](NodeID node) {
+            auto intersection = intersection_generator(node, sliproad_edge);
+            // skip over traffic lights
+            if (intersection.size() == 2)
+            {
+                node = node_based_graph.GetTarget(sliproad_edge);
+                sliproad_edge = intersection[1].eid;
+                intersection = intersection_generator(node, sliproad_edge);
+            }
+            return intersection;
+        }(intersection_node_id);
+
+        const NodeID sliproad_edge_target = node_based_graph.GetTarget(sliproad_edge);
```

Comment:
you now have `d` and `sliproad_edge_target` both from `sliproad_edge`. Following these without better names can become difficult. I'd prefer `first_node_along_sliproad_candidate` and `node_at_next_intersection_along_sliproad`